pkg/apis/provisioning/v1alpha5/constraints.go

```diff
@@ -48,27 +48,22 @@ func (c *Constraints) ValidatePod(pod *v1.Pod) error {
 	if err := c.Taints.Tolerates(pod); err != nil {
 		return err
 	}
-	// The constraints do not support this requirement
-	podRequirements := PodRequirements(pod)
-	for _, key := range podRequirements.Keys() {
-		if c.Requirements.Requirement(key).Len() == 0 {
-			return fmt.Errorf("invalid nodeSelector %q, %v not in %v", key, podRequirements.Requirement(key).UnsortedList(), c.Requirements.Requirement(key).UnsortedList())
-		}
+	// Test if pod requirements are valid
+	podRequirements := NewRequirements(PodRequirements(pod)...)
```

Comment:
🤔 how could the pod requirements be invalid? We should have validated them before we even get to provisioning.

Reply:
I think we probably want to do this out in the selection controller.

Reply:
What does it look like for the pod's requirements to be invalid, anyways? Is it possible for them to be invalid, but the compatibility check to succeed?